completion/available/system.completion.bash

```diff
@@ -3,26 +3,43 @@
 # Loads the system's Bash completion modules.
 # If Homebrew is installed (OS X), its Bash completion modules are loaded.
 
-if [ -f /etc/bash_completion ]; then
-  . /etc/bash_completion
+if [ -r /etc/bash_completion ] ; then
+  # shellcheck disable=SC1091
+  source /etc/bash_completion
 fi
 
 # Some distribution makes use of a profile.d script to import completion.
-if [ -f /etc/profile.d/bash_completion.sh ]; then
-  . /etc/profile.d/bash_completion.sh
+if [ -r /etc/profile.d/bash_completion.sh ] ; then
+  # shellcheck disable=SC1091
+  source /etc/profile.d/bash_completion.sh
 fi
 
 
-if [ $(uname) = "Darwin" ] && command -v brew &>/dev/null ; then
-  BREW_PREFIX=$(brew --prefix)
+if [ "$(uname)" == "Darwin" ] ; then
 
-  if [ -f "$BREW_PREFIX"/etc/bash_completion ]; then
-    . "$BREW_PREFIX"/etc/bash_completion
+  # XCode installs git's completion files to a non-standard place
+  # Note: This should be loaded before brew's completion folder is parsed,
+  #       multiple tools glob on to this completion.
+  _git_comp='/Applications/Xcode.app/Contents/Developer/usr/share/git-core/git-completion.bash'
```

Comment:
I'm not sure I like the fact that we're including git-specific checks in here, but on the other hand, it makes sense to keep things in one place - we'll just have to make sure that we follow that pattern consistently.

Reply:
@nwinkler Yeah I'm honestly torn on this and leave the decision up to you. I can totally put this back into `git.completion.bash`, but I decided on this because it's installed with the "system", just outside of the normal locations. I did some searching around in Xcode and it seems like this is the only case where this happens.

Reply:
I think it's fine to leave this here - there are comments explaining why we have this, should be good.